CRM/Utils/Geocode/Google.php

```diff
@@ -82,7 +82,7 @@ public static function format(&$values, $stateName = FALSE) {
       $add .= ',+';
     }
 
-    if (!empty($values['state_province']) || !empty($values['state_province_id'])) {
+    if (!empty($values['state_province']) || (!empty($values['state_province_id']) && $values['state_province_id'] != 'null')) {
```

Comment:
I think it is explained in the Ticket I think it is when you deselect the state from an address